packages/fxa-content-server/app/scripts/views/subscriptions_product_redirect.js

```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import FormView from './form';
+import Template from 'templates/subscriptions_redirect.mustache';
+import PaymentServer from '../lib/payment-server';
+
+class SubscriptionsProductRedirectView extends FormView {
+  mustAuth = true;
+  template = Template;
+
+  initialize (options) {
+    this._currentPage = options.currentPage;
+    this._subscriptionsConfig = {};
+    if (options && options.config && options.config.subscriptions) {
+      this._subscriptionsConfig = options.config.subscriptions;
+    }
+  }
+
+  afterRender () {
+    const searchQuery = this.window.location.search;
```

Comment:
Do we want to keep the query string? The whole redirecting-with-token thing makes me a bit nervous, and in this case passing through information also makes me nervous. (Though I'm guessing very soon we'll have utm parameters that we would want to pass through right here.)

I might be just paranoid here, I'm not sure.

Reply:
Yeah, there will be a laundry list of query params we'll want to pass through - including plan ID for a product, GA params, etc. We could whitelist them, but it would probably be an ever-growing list. I can't think of a param that would be harmful per se, though

Reply:
I'd be much more comfortable going with an explicit allow list *and* validating their form on the subscription service to minimize the possibility of XSS. Perhaps the validation code from the content server could be re-used by the payments server, or maybe those params could be valdidated on the backend to reduce the amount of logic shipped to the front end (in retrospect, this is the approach I wish we would have done on the content server).